src/test/e2e/05_tarball_test.go

```diff
@@ -36,6 +37,21 @@ func TestMultiPartPackage(t *testing.T) {
 	require.NoError(t, err)
 	// Length is 7 because there are 6 parts and 1 manifest
 	require.Len(t, parts, 7)
+	// Check the file sizes are even
+	part1FileInfo, err := os.Stat(parts[1])
+	require.NoError(t, err)
+	require.Equal(t, int64(1000000), part1FileInfo.Size())
+	part2FileInfo, err := os.Stat(parts[2])
+	require.NoError(t, err)
+	require.Equal(t, int64(1000000), part2FileInfo.Size())
+	// Check the package data is correct
+	pkgData := types.ZarfSplitPackageData{}
+	part0File, err := os.ReadFile(parts[0])
+	require.NoError(t, err)
+	err = json.Unmarshal(part0File, &pkgData)
+	require.NoError(t, err)
+	require.Equal(t, pkgData.Count, 6)
+	fmt.Printf("%#v", pkgData)
```

Comment:
Just so you are aware, the original code would pass this test since the buffer size will dynamically change if it is is below 16MiB. Only when --max-package-size > 17 does the error show up.

Reply:
Good point! updated